tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -393,33 +411,57 @@ def _test_status_check_replacement(
-            "partition": partition,
-            "constraint": "dynamic",
-            "other_options": "-a 1-{} --exclusive --no-requeue".format(num_dynamic_nodes),
-        }
+    # Assert that the default SlurmdTimeout=180 is in effect
+    _assert_slurmd_timeout(remote_command_executor, timeout=180)
+    # Nodes with networking failures should fail slurm health check before failing ec2_status_check
+    # Test on freshly launched dynamic nodes
+    kill_job_id = _submit_kill_networking_job(
+        remote_command_executor, scheduler_commands, partition, node_type="dynamic", num_nodes=num_dynamic_nodes
+    )
+    time.sleep(15)
+    scheduler_commands.cancel_job(kill_job_id)
+    # Assert dynamic nodes are in DOWN
+    _wait_for_compute_nodes_states(scheduler_commands, dynamic_nodes, expected_states=["down*", "down"])
+    # Assert dynamic nodes are reset
+    _wait_for_node_reset(scheduler_commands, static_nodes=[], dynamic_nodes=dynamic_nodes)
+    assert_num_instances_in_cluster(cluster_name, region, len(static_nodes))
+    # Assert ec2_status_check code path is not triggered
+    assert_no_msg_in_logs(
+        remote_command_executor,
+        ["/var/log/parallelcluster/clustermgtd"],
+        ["Setting nodes failing health check type ec2_health_check to DRAIN"],
     )
-    # Assert that dynamic nodes with failing status check are put into DRAIN
-    # Can take up to 15 mins for status check to show
+
+
+def _test_status_check_replacement(
+    remote_command_executor,
+    scheduler_commands,
+    cluster_name,
+    region,
+    partition,
+    num_static_nodes,
+):
+    """Test nodes with failing ec2 status checks are correctly replaced."""
+    logging.info("Testing that nodes with failing ec2 status checks are correctly replaced")
+    static_nodes, _ = assert_initial_conditions(scheduler_commands, num_static_nodes, 0, partition)
+    # Can take up to 15 mins for ec2_status_check to show
+    # Need to increase SlurmdTimeout to avoid slurm health check and trigger ec2_status_check code path
+    _set_slurmd_timeout(remote_command_executor, timeout=10000)
+    kill_job_id = _submit_kill_networking_job(
+        remote_command_executor, scheduler_commands, partition, node_type="static", num_nodes=num_static_nodes
+    )
+    # Assert ec2_status_check code path is triggered
     retry(wait_fixed=seconds(20), stop_max_delay=minutes(15))(assert_errors_in_logs)(
         remote_command_executor,
         ["/var/log/parallelcluster/clustermgtd"],
         ["Setting nodes failing health check type ec2_health_check to DRAIN"],
     )
     scheduler_commands.cancel_job(kill_job_id)
-    # Assert dynamic nodes are reset
-    _wait_for_node_reset(scheduler_commands, static_nodes=[], dynamic_nodes=dynamic_nodes)
+    # Assert static nodes are reset
+    _wait_for_node_reset(scheduler_commands, static_nodes=static_nodes, dynamic_nodes=[])
     assert_num_instances_in_cluster(cluster_name, region, len(static_nodes))
```

Comment:
do we need to check logs to make sure the replacement happened due to failing health checks?

Reply:
Yes, we check that `Setting nodes failing health check type ec2_health_check to DRAIN` is in clustermgtd log above